src/vs/platform/terminal/common/xterm/shellIntegrationAddon.ts

```diff
@@ -333,14 +333,12 @@ export class ShellIntegrationAddon extends Disposable implements IShellIntegrati
 				return true;
 			}
 			case VSCodeOscPt.CommandLine: {
-				let commandLine: string;
-				if (args.length >= 1 || args.length <= 2) {
-					commandLine = deserializeMessage(args[0]);
-				} else {
-					commandLine = '';
+				const commandLine: string | undefined = args.length ? deserializeMessage(args[0]) : undefined;
```

Comment:
I think this would still fail as args is undefined? We want the type of `args` on line 319 to be `string[] | undefined`

Reply:
![Screenshot 2023-05-09 at 12 40 32 PM](https://github.com/microsoft/vscode/assets/29464607/15b8c3b7-70ca-4e98-b6b1-e1beb4c110b0)

Reply:
This is the problem:

![image](https://github.com/microsoft/vscode/assets/2193314/dec1a05a-26ef-48be-8b0e-b98c27032cb5)

Reply:
I feel like this function could really use some tests as I'm not 100% sure what's expected for each case